worldedit-bukkit/adapters/adapter-1.17.1/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/v1_17_R1_2/PaperweightAdapter.java

```diff
@@ -500,28 +500,34 @@ public Component getRichItemName(BaseItemStack itemStack) {
     }
 
     @SuppressWarnings({ "unchecked", "rawtypes" })
-    @Override
-    public Map<String, ? extends Property<?>> getProperties(BlockType blockType) {
-        Map<String, Property<?>> properties = Maps.newTreeMap(String::compareTo);
-        Block block = getBlockFromType(blockType);
-        StateDefinition<Block, net.minecraft.world.level.block.state.BlockState> blockStateList =
-            block.getStateDefinition();
-        for (net.minecraft.world.level.block.state.properties.Property state : blockStateList.getProperties()) {
-            Property property;
+    private static final LoadingCache<net.minecraft.world.level.block.state.properties.Property, Property<?>> propertyCache = CacheBuilder.newBuilder().build(new CacheLoader<net.minecraft.world.level.block.state.properties.Property, Property<?>>() {
```

Comment:
As a `private static final` this should be `PROPERTY_CACHE` IMO. Same for the others.